analyzers/src/SonarAnalyzer.CSharp/Extensions/SyntaxNodeExtensions.cs

```diff
@@ -31,18 +31,21 @@ internal static partial class SyntaxNodeExtensions
             SyntaxKind.AddAccessorDeclaration,
             SyntaxKind.AnonymousMethodExpression,
             SyntaxKind.BaseConstructorInitializer,
+            SyntaxKind.CompilationUnit,
             SyntaxKind.ConstructorDeclaration,
             SyntaxKind.ConversionOperatorDeclaration,
             SyntaxKind.DestructorDeclaration,
-            SyntaxKind.EqualsValueClause,
```

Comment:
EqualsValueClause is to broad. It is used in all kinds of contexts
```cs
int field = 1;
var local = 1;
M(int parameter = 1) { }
int Property { get; } = 1;
enum E { EnumMember = 1 }
```

For the local variable declaration, it is wrong. I replaced it by the other kinds.